docs/reference/builder.md

```diff
@@ -109,22 +109,169 @@ Here is the format of the `Dockerfile`:
+before any instruction in a Dockerfile, and may be included once, and once only,
+in a Dockerfile. Parser directives are written as a special type of comment in
+the form `# DIRECTIVE=value`.  Line continuation characters are not supported in
+parser directives. Hence the following examples are all invalid:
+
+    # Invalid due to line continuation
+    # DIREC \
+    TIVE=value
+or
+
+    # Invalid due to appearing twice
+    # DIRECTIVE=value1
+    FROM ImageName
+    # DIRECTIVE=value2
+    
+or
+
+    # Invalid due to appearing after a builder instruction
+    FROM ImageName
+    # DIRECTIVE=value
+
+The following would be treated as a comment due to the directive not being
+recognised:
+
+    # UNRECOGNISEDDIRECTIVE=value    
+    
+	
+Directives are not case-sensitive.  However, convention is for them to
+be UPPERCASE to distinguish them from other comments more easily. Non 
+line-breaking whitespace is permitted, as is a subsequent comment. Hence 
+these examples are all syntactically valid, and will be treated identically:
+
+    #DIRECTIVE=value
+    # DIRECTIVE =value
+	#	DIRECTIVE= value		# Some ignored comment
+	# DIRECTIVE = value
+	#	  dIrEcTiVe=value
+
+The following parser directive is supported:
+
+* `ESCAPE`
+
+## ESCAPE
+
+    # ESCAPE=\
+
+Or
+
+    # ESCAPE=`
+
+`ESCAPE` sets the character used to escape characters in a `Dockerfile`. 
+If not specified, the default escape character is `\`. 
+
+The escape character is used both to escape characters in a line, and to
+escape a newline. This allows a `Dockerfile` instruction to
+span multiple lines. Note that regardless of whether the `ESCAPE` parser
+directive is included in a `Dockerfile`, *escaping is not performed in 
+a `RUN` command, except at the end of a line.* 
+
+Setting the escape character to `` ` `` is especially useful on 
+`Windows`, where `\` is the directory path separator. `` ` `` is consistent 
+with [Windows PowerShell](https://technet.microsoft.com/en-us/library/hh847755.aspx).
+
+Consider the following example which would fail in a non-obvious way on 
+`Windows`. The second `\` at the end of the second line would be interpreted as an
+escape for the newline, instead of a target of the escape from the first `\`. 
+Similarly, the `\` at the end of the third line would, assuming it was actually
+handled as an instruction, cause it be treated as a line continuation. The result
+of this dockerfile is that second and third lines are considered a single
+instruction: 
+
+    FROM windowsservercore
+    COPY testfile.txt c:\\
+    RUN dir c:\
+
+Results in:
+
+    PS C:\John> docker build -t cmd .
+    Sending build context to Docker daemon 3.072 kB
+    Step 1 : FROM windowsservercore
+     ---> dbfee88ee9fd
+    Step 2 : COPY testfile.txt c:\ RUN dir c:
+    Forbidden path outside the build context: c: ()
+    PS C:\John>	
+
+Similarly, the following would also fail, but in a slightly different way:
```

Comment:
Can you please explain what is different between the 2 examples - they look identical!